SingularityService/src/main/java/com/hubspot/singularity/data/history/DeployTaskHistoryHelper.java

```diff
@@ -0,0 +1,112 @@
+package com.hubspot.singularity.data.history;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import com.google.common.base.Predicate;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Lists;
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import com.hubspot.singularity.SingularityDeployKey;
+import com.hubspot.singularity.SingularityTask;
+import com.hubspot.singularity.SingularityTaskHistoryUpdate;
+import com.hubspot.singularity.SingularityTaskId;
+import com.hubspot.singularity.SingularityTaskIdHistory;
+import com.hubspot.singularity.config.SingularityConfiguration;
+import com.hubspot.singularity.data.TaskManager;
+
+@Singleton
+public class DeployTaskHistoryHelper extends BlendedHistoryHelper<SingularityTaskIdHistory, SingularityDeployKey> {
+
+  private final TaskManager taskManager;
+  private final HistoryManager historyManager;
+  private final SingularityConfiguration singularityConfiguration;
+
+  @Inject
+  public DeployTaskHistoryHelper(TaskManager taskManager, HistoryManager historyManager, SingularityConfiguration singularityConfiguration) {
+    this.taskManager = taskManager;
+    this.historyManager = historyManager;
+    this.singularityConfiguration = singularityConfiguration;
+  }
+
+  public List<SingularityTaskIdHistory> getHistoriesFor(Collection<SingularityTaskId> taskIds) {
```

Comment:
This method should not be public since it is not used publicly. Also looks like a direct copy/paste from TaskHistoryHelper. If so, it should either use a library function or inherit the function.